src/PhotoNim.nim

```diff
@@ -80,8 +105,92 @@ if args["pfm2png"]:
     
     pfm2png(fileIn, fileOut, alpha, gamma)
 
+#-----------------------------#
+#       Demo executable       #
+#-----------------------------#
+
 elif args["demo"]: 
-    quit "Eh volevi"
+    var 
+        height = 1000
+        width = 1600
+
+    if args["--width"]: 
+        try: width = parseInt($args["--width"]) 
+        except: echo "Warning: width must be an integer. Default value is used."
+    
+    if args["--height"]: 
+        try: height = parseInt($args["--height"]) 
+        except: echo "Warning: height must be an integer. Default value is used."
+
+    
+    #----------------------------------------------#
+    #              Defining scenary                #
+    #----------------------------------------------#
+
+    let 
+        timeStart = cpuTime()
+        a_ratio = float32(width)/float32(height)
+        sc = newScaling(0.1)    # Scaling needed in order to have 1/10 radius -> we will compose it with s translation
+        s1 = newSphere(newTranslation(newVec3[float32](0.5, 0.5, 0.5)) @ sc)
+        s2 = newSphere(newTranslation(newVec3[float32](0.5, 0.5, -0.5)) @ sc)
+        s3 = newSphere(newTranslation(newVec3[float32](0.5, -0.5, 0.5)) @ sc)
+        s4 = newSphere(newTranslation(newVec3[float32](0.5, -0.5, -0.5)) @ sc)
+        s5 = newSphere(newTranslation(newVec3[float32](-0.5, 0.5, 0.5)) @ sc)
+        s6 = newSphere(newTranslation(newVec3[float32](-0.5, 0.5, -0.5)) @ sc)
+        s7 = newSphere(newTranslation(newVec3[float32](-0.5, -0.5, 0.5)) @ sc)
+        s8 = newSphere(newTranslation(newVec3[float32](-0.5, -0.5, -0.5)) @ sc)
+        s9 = newSphere(newTranslation(newVec3[float32](-0.5, 0.0, 0.0)) @ sc)
+        s10 = newSphere(newTranslation(newVec3[float32](0.0, 0.5, 0.0)) @ sc)   
+ 
+    var 
+        image = newHdrImage(width, height)
+        rotz: Rotation
+        cam: Camera
+        ang: int = 10
+
+    if args["--angle"]:
+        try: ang = parseInt($args["--angle"]) 
+        except: echo "Warning: angle must be an integer. Default value is used."
+    
+    rotz = newRotZ(float32(ang))
+    
+    if args["perspective"]:
```

Comment:
Here, if the user pass "perspecteve" the program should fail.

Reply:
I'm not entirely sure about what you mean, but if the user pass perspecteve program execution doesn't even start:
![Screenshot from 2024-05-11 20-18-54](https://github.com/Negrini085/PhotoNim/assets/139368862/c5b2d99d-caec-441f-a9d7-26447bfebdf1)

Reply:
If you could pass something less verbose to write (p for perspective, o for orthogonal) it would be easier.